core/src/test/java/org/apache/calcite/sql/test/SqlTestFactory.java

```diff
@@ -133,11 +133,20 @@ public SqlValidator getValidator() {
         (SqlConformance) options.get("conformance");
     final boolean lenientOperatorLookup =
         (boolean) options.get("lenientOperatorLookup");
-    final boolean enableTypeCoercion = (boolean) options.get("enableTypeCoercion");
+    final boolean enableTypeCoercion =
+        (boolean) options.get("enableTypeCoercion");
+    final boolean handleUnknownTables;
+    if (options.containsKey("handleUnknownTables")) {
```

Comment:
The other options don't seen to check for existence before calling get(). Can we just call get() directly? Will it default to returning false if the key is not found?

Reply:
It will fail if we call get() on a nonexistent key. I believe the other options are not checked beforehand because they are always set explicitly. I figured it would be easiest to have handleUnknownTables be optional and just check that it is in the map before retrieving it, but I could change the existing code to always set the handleUnknownTables option explicitly to either true or false if you think that is preferable.

Reply:
To keep things consistent, let's follow the same pattern as is used for all of the other options.